src/utils/RateLimitedV2.sol

```diff
@@ -128,7 +149,7 @@ abstract contract RateLimitedV2 is IRateLimitedV2, CoreRefV2 {
         emit BufferReplenished(amount, bufferStored);
```

Comment:
also use `newBufferStored` when you emit `BufferReplenished` to save a warm sload (same logic as the change you did for `_depleteBuffer`)